draft-ietf-ppm-dap.md

```diff
@@ -1109,32 +1109,37 @@ aggregator (`0x02` for the leader and `0x03` for the helper). If decryption
 fails, the aggregator marks the report share as invalid with the error
 `hpke-decrypt-error`. Otherwise, it outputs the resulting `input_share`.
 
-#### Input Share Validation {#input-share-batch-validation}
+#### Early Input Share Validation {#early-input-share-validation}
 
 Validating an input share will either succeed or fail. In the case of failure,
 the input share is marked as invalid with a corresponding ReportShareError error.
 
-The validation checks are as follows.
+Before beginning the preparation step, Aggregators are required to perform the
+following generic checks.
 
 1. Check if the report has never been aggregated but is contained by
    a batch that has been collected. If this check fails, the input share
    MUST be marked as invalid with the error `batch-collected`. This prevents
    additional reports from being aggregated after its batch has already
    been collected.
-2. Check if the report has already been aggregated with this aggregation
+1. Check if the report has already been aggregated with this aggregation
    parameter. If this check fails, the input share MUST be marked as invalid
    with the error `report-replayed`. This is the case if the report was used in
-   a previous aggregate request and is therefore a replay. An aggregator may
-   also choose to mark an input share as invalid with the error
-   `report-dropped` under the conditions prescribed in {{anti-replay}}.
-3. Depending on the query type for the task, additional checks may be applicable:
+   a previous aggregate request and is therefore a replay.
+1. Depending on the query type for the task, additional checks may be applicable:
     * For fixed-size tasks, the Aggregators need to ensure that each batch is
       roughly the same size. If the number of reports aggregated for the current
       batch exceeds the maximum batch size (per the task configuration), the
       Aggregator MAY mark the input share as invalid with the error
       "batch-saturated". Note that this behavior is not strictly enforced here
       but during the collect sub-protocol. (See {{batch-validation}}.)
-If both checks succeed, the input share is not marked as invalid.
+1. Finally, if an Aggregator cannot determine if an input share is valid. it
+   MUST mark the input share as invalid with error `report-dropped`. This
+   situation arises if, for example, the Aggregator has evicted from long-term
+   storage the state required to perform the check. (See
```

Comment:
This would also happen if for instance a helper received an `AggregateContinueReq` that contained a `PrepareStep` referencing a report that the helper has never seen before, right? If so, maybe worth noting as much in the "Helper Continuation" section that discusses handling of `AggregateContinueReq`.

Reply:
That's not really the situation I'm thinking of. I suspect what I'm imagining will be much more common: Suppose a task runs for year. An Aggregator may choose to evict all report metadata that is older than a week; this text is meant to cover the situation where a report arrives over a week late.